crates/components-runtime/src/data_collection/mod.rs

```
@@ -78,6 +85,96 @@ pub async fn send_json_events(
+        )
+        .await
+        .unwrap_or(false),
+    )
+}
+
+pub async fn get_auth_request(
+    context: &ComponentsContext,
+    component: &DataCollectionComponents,
+) -> anyhow::Result<JoinHandle<AuthResponse>> {
+    let mut store = context.empty_store();
+    let (headers, method, url, body, auth_metadata) =
+        match crate::data_collection::versions::v1_0_1::execute::get_auth_request(
+            context, component, &mut store,
+        )
+        .await
+        {
+            Ok((headers, method, url, body, auth_metadata)) => {
+                (headers, method, url, body, auth_metadata)
+            }
+            Err(err) => {
+                error!("Failed to get auth request. Error: {}", err);
+                return Err(err);
+            }
+        };
+
+    let client = reqwest::Client::builder()
+        .timeout(Duration::from_secs(5))
+        .build()?;
+
+    let component_id = component.id.clone();
+    let future = tokio::spawn(async move {
+        let res = match method.as_str() {
+            "GET" => client.get(url.clone()).headers(headers).send().await,
+            "PUT" => {
+                client
+                    .put(url.clone())
+                    .headers(headers)
+                    .body(body.clone())
+                    .send()
+                    .await
+            }
+            "POST" => {
+                client
+                    .post(url.clone())
+                    .headers(headers)
+                    .body(body.clone())
+                    .send()
+                    .await
+            }
+            _ => {
+                return AuthResponse {
+                    component_id,
+                    token: String::new(),
+                    token_duration: auth_metadata.token_duration,
+                }
+            }
+        };
+
+        let response_text = match res {
+            Ok(r) => r.text().await.unwrap_or_default(),
+            Err(_) => String::new(),
```

should we log this somewhere too? failing silently is (almost) always a bad idea